pkg/client/unversioned/request.go

```diff
@@ -232,13 +234,27 @@ func (r *Request) AbsPath(segments ...string) *Request {
 	if r.err != nil {
 		return r
 	}
+	r.path = absPath(r.serverBase, segments...)
+	return r
+}
+
+func absPath(base string, segments ...string) string {
+	// special case: if base ends in /api then the "real" relative server base is ..
```

Comment:
Why?  This seems like a really strange thing to do.  Why not choose the base you mean?